### ISSUE TITLE
Add code-server as IDE option

### ISSUE THREAD

**Issue by sr229** [JanitorTechnology/janitor#470]
[code-server](https://github.com/cdr/code-server) builds upon the open source [vscode](https://github.com/microsoft/vscode) to make it accessible for the web.

We should be able to provide this an option since its more flexible the Cloud9 at major parts. This'll also improve developer UX since we can use VSCode extensions in place of custom Cloud9 plugins (for example, the C++ extension has good IntelliSense that'll help out Chromium developers know what's wrong immediately via Clang).

*Note: We should be waiting for the `web` branch to merge since we can just change the route where code-server can serve properly. You may try the `web` branch feature [here](https://github.com/sr229/cdr-web-lkgr/releases).

**Comment by etiennewan** [comment]
Good idea. We already got Theia IDE and Cloud9 IDE running in the environments (though we do not use Theia IDE as the default right now). We can look to add code-server in the ubuntu-dev dockerfile as a new alternative.
We should leave the choice of the IDE to the user at some point, that would mean we would have to choose a default option for the IDE. (relevant for a future discussion)
But for the moment, I think Cloud9 IDE is the only one that supports live collaboration, which is a feature we would need if we enable sharing containers between users. If code-server has it too, then code-server would be a good candidate.

**Comment by sr229** [comment]
@etiennewan judging from `web` branch's source, it uses the web version of VSCode so Microsoft Live Share is supposed to work.

**Comment by sr229** [comment]
And further digging, we still need to point the VSCode extension URLs to Microsoft's to get the official extensions
